plugins/scaffolder-backend/src/lib/templating/filters.ts

```diff
@@ -15,26 +15,186 @@
  */
 import { parseEntityRef } from '@backstage/catalog-model';
 import { ScmIntegrations } from '@backstage/integration';
-import type { JsonObject, JsonValue } from '@backstage/types';
 import {
+  CreatedTemplateFilter,
+  createTemplateFilter,
   parseRepoUrl,
   TemplateFilter,
 } from '@backstage/plugin-scaffolder-node';
+import type { JsonObject, JsonValue } from '@backstage/types';
+import { keyBy, mapValues } from 'lodash';
 import get from 'lodash/get';
+import { z } from 'zod';
+
+const parseRepoUrlFilter = (integrations: ScmIntegrations) => {
+  return createTemplateFilter({
+    id: 'parseRepoUrl',
+    description:
+      'Parses a repository URL into its components, such as owner, repository name, and more.',
+    schema: {
+      input: z
+        .string()
+        .describe('repo URL as collected from repository picker'),
+      output: z
+        .object({
+          repo: z.string(),
+          host: z.string(),
+        })
+        .merge(
+          z
+            .object({
+              owner: z.string(),
+              organization: z.string(),
+              workspace: z.string(),
+              project: z.string(),
+            })
+            .partial(),
+        )
+        .describe('`RepoSpec`'),
+    },
+    examples: [
+      {
+        example: `- id: log
+name: Parse Repo URL
+action: debug:log
+input:
+extra: \${{ parameters.repoUrl | parseRepoUrl }}`,
+        notes: ` - **Input**: \`github.com?repo=backstage&owner=backstage\`
+- **Output**: \`{"host":"github.com","owner":"backstage","repo":"backstage"}\`
+`,
+      },
+    ],
+    impl: url => parseRepoUrl(url as string, integrations),
```

Comment:
We should be able to type the all the arguments to the `impl` function by using `z.output<InputType>` and `z.output<ArgumentsType>` I think, and then we can do validation in the filter wrapping to ensure that's what the filters get.

Reply:
I'm also thinking the we should rename `impl` to something like `filter` or `fn` instead.

Reply:
I'd lean toward `filter` to avoid confusion with global functions... _or_ should we consider that filters are functions and make a given such function available for invocation either as filter or function? Is that giving away too much of the implementation?